ocp_resources/catalog_source.py

```diff
@@ -1,34 +1,32 @@
-from ocp_resources.constants import TIMEOUT_4MINUTES
 from ocp_resources.resource import NamespacedResource
 
 
 class CatalogSource(NamespacedResource):
+    """
+    https://olm.operatorframework.io/docs/concepts/crds/catalogsource/
+    """
+
     api_group = NamespacedResource.ApiGroup.OPERATORS_COREOS_COM
 
     def __init__(
         self,
-        name=None,
-        namespace=None,
-        client=None,
         source_type=None,
         image=None,
         display_name=None,
         publisher=None,
-        teardown=True,
-        yaml_file=None,
-        delete_timeout=TIMEOUT_4MINUTES,
         update_strategy_registry_poll_interval=None,
         **kwargs,
     ):
-        super().__init__(
-            name=name,
-            namespace=namespace,
-            client=client,
-            teardown=teardown,
-            yaml_file=yaml_file,
-            delete_timeout=delete_timeout,
-            **kwargs,
-        )
+        """
+        Args:
+            source_type (str): Name of the source type.
+            image (str): Image index for the catalog.
+            display_name (str): Display name for the catalog in the web console and CLI.
+            publisher (str): Name of the publisher.
+            update_strategy_registry_poll_interval (str): Time interval between checks of the latest
```

Comment:
```suggestion
            update_strategy_registry_poll_interval (str, optional): Time interval between checks of the latest
```

please indicate in all places args that are optional (for those that are not optional and we do not have a validation about them in to_dict, please open issues to fix)